src/pages/HomeParticipant.vue

```diff
@@ -375,23 +376,20 @@ const studentInfo = computed(() => {
   };
 });
 
-watch(consentParams, (newValue) => {
-  consentParams.value = newValue;
-});
-
 watch(
   selectedAdmin,
-  async (newValue) => {
-    if (newValue) {
-      await checkConsent();
-    }
+  () => {
+    checkConsent();
   },
   { immediate: true },
 );
 
 watch(
   adminInfo,
   () => {
+    if (selectedAdmin.value) {
+      checkConsent();
+    }
```

Comment:
I think that this can be simplified into a single watcher.

```suggestion
    watch([selectedAdmin, adminInfo], ([updateSelectedAdmin, updateAdminInfo]) => {
        if (updateSelectedAdmin) {
            checkConsent()
        }
    }, {immediate: true}
    );
```

Reply:
totally, thank you for the suggestion, I will apply